libyul/optimiser/SimplificationRules.h

```diff
@@ -42,37 +44,6 @@ class Pattern;
 /**
  * Container for all simplification rules.
  */
-class SimplificationRules: public boost::noncopyable
```

Comment:
Why did you move the class?

Reply:
Mainly because of the current usage of the `Pattern` class. Because of the use of forward declaration `Pattern` class was of incomplete type. However, I could change it to a pointer.